instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/splunkkafka/test/kafka_test.go

```diff
@@ -0,0 +1,212 @@
+		t.Skip("skipping test in short mode.")
+	}
+	// Test consuming via the Events channel by artificially sending messages.
+
+	sr := tracetest.NewSpanRecorder()
+	tp := trace.NewTracerProvider(trace.WithSpanProcessor(sr))
+
+	c, err := splunkkafka.NewConsumer(&kafka.ConfigMap{
+		"go.events.channel.enable": true, // required for the events channel to be turned on
+		"group.id":                 testGroupID,
+		"session.timeout.ms":       10,
+		"enable.auto.offset.store": false,
+		// "debug":                    "all",
+	}, splunkkafka.WithTracerProvider(tp))
+	require.NoError(t, err)
+
+	err = c.Subscribe(testTopic, nil)
+	require.NoError(t, err)
+
+	go func() {
+		c.Consumer.Events() <- &kafka.Message{
+			TopicPartition: kafka.TopicPartition{
+				Topic:     &testTopic,
+				Partition: 1,
+				Offset:    1,
+			},
+			Key:   []byte("key1"),
+			Value: []byte("value1"),
+		}
+		c.Consumer.Events() <- &kafka.Message{
+			TopicPartition: kafka.TopicPartition{
+				Topic:     &testTopic,
+				Partition: 1,
+				Offset:    2,
+			},
+			Key:   []byte("key2"),
+			Value: []byte("value2"),
+		}
+	}()
+
+	msg1 := (<-c.Events()).(*kafka.Message)
+	assert.Equal(t, []byte("key1"), msg1.Key)
+	msg2 := (<-c.Events()).(*kafka.Message)
+	assert.Equal(t, []byte("key2"), msg2.Key)
+
+	c.Close()
+	// wait for the events channel to be closed
+	<-c.Events()
+
+	spans := sr.Ended()
+	assert.Len(t, spans, 2)
+	expectedName := fmt.Sprintf("%s receive", testTopic)
+	for i, s := range spans {
+		assert.Equal(t, expectedName, s.Name())
+		assert.Equal(t, traceapi.SpanKindConsumer, s.SpanKind())
+		attrs := s.Attributes()
+		assert.Contains(t, attrs, semconv.MessagingSystemKey.String("kafka"))
+		assert.Contains(t, attrs, semconv.MessagingDestinationKindTopic)
+		assert.Contains(t, attrs, semconv.MessagingDestinationKey.String(testTopic))
+		assert.Contains(t, attrs, semconv.MessagingOperationReceive)
+		offset := kafka.Offset(i + 1)
+		assert.Contains(t, attrs, semconv.MessagingMessageIDKey.String(offset.String()))
+		key := fmt.Sprintf("key%d", i+1)
+		assert.Contains(t, attrs, semconv.MessagingKafkaMessageKeyKey.String(key))
+		assert.Contains(t, attrs, semconv.MessagingKafkaConsumerGroupKey.String(testGroupID))
+		assert.Contains(t, attrs, semconv.MessagingKafkaPartitionKey.Int64(1))
+	}
+}
+
+/*
+to run the integration test locally:
+    docker network create confluent
+    docker run --rm \
+        --name zookeeper \
+        --network confluent \
+        -p 2181:2181 \
+        -e ZOOKEEPER_CLIENT_PORT=2181 \
+        confluentinc/cp-zookeeper:5.0.0
+    docker run --rm \
+        --name kafka \
+        --network confluent \
+        -p 9092:9092 \
+        -e KAFKA_ZOOKEEPER_CONNECT=zookeeper:2181 \
+        -e KAFKA_ADVERTISED_LISTENERS=PLAINTEXT://localhost:9092 \
+        -e KAFKA_LISTENERS=PLAINTEXT://0.0.0.0:9092 \
+        -e KAFKA_CREATE_TOPICS=gotest:1:1 \
+        -e KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1 \
+        confluentinc/cp-kafka:5.0.0
+*/
```

Comment:
I prefer to use something like https://github.com/ory/dockertest

The reasons are:
- no need to create special Make targets. one can just use `go test` and it works. We could remove the `INTEGRATION` env var.
- does not require Docker CLI so often you can run such tests inside a `golang` docker container if you just mount `-v /var/run/docker.sock:/var/run/docker.sock` (here I am not sure if it will work b/c of native dependencies)

Reply:
I just learned that this approach is even recommended by Docker 😉 
See: https://docs.docker.com/language/golang/run-tests/

Reply:
I think it is not worth addressing it in this PR.
I created an issue out of it: https://github.com/signalfx/splunk-otel-go/issues/101
I would be thankful if you express your opinion there 😉

Feel free to resolve this comment.